src/s3/S3Agent.ts

```diff
@@ -89,29 +94,58 @@ class S3Agent {
     return counter;
   }
 
-  public async rejectAvatarRequest(link: string) {
-    const pattern = new RegExp(`${FolderName.PENDING}\/.*$`);
-    if (!link.match(pattern)) {
-      throw new Error(`Invalid link provided: ${link}`);
+  public async rejectAvatarRequest(s3AvatarLinks: S3AvatarLinks) {
+    if (!this.isValidPendingAvatarRequest(s3AvatarLinks)) {
+      throw new Error(
+        `Invalid links provided: resLink="${s3AvatarLinks.resLink}" spyLink="${s3AvatarLinks.spyLink}"`,
+      );
```

Comment:
As discussed, let's try first and ask for forgiveness afterwards.

Reply:
Function is for determining processing avatar requests are only done on pending avatars, not other files